cypress/e2e/spec-wc.cy.js

```diff
@@ -142,3 +142,58 @@ describe("when load_remix_disabled is true, e.g. in editor-standalone", () => {
     });
   });
 });
+
+describe("when embedded, output_only & output_split_view are true", () => {
+  const urlFor = (identifier) => {
+    const params = new URLSearchParams();
+    params.set("identifier", identifier);
+    params.set("load_remix_disabled", "true");
+    params.set("embedded", "true");
+    params.set("output_only", "true");
+    params.set("output_split_view", "true");
+    return `${origin}?${params.toString()}`;
+  };
+
+  it("displays the embedded view for a Python project", () => {
+    cy.visit(urlFor("clean-car-example"));
+
+    // Check text output panel is visible and has a run button
+    // Important to wait for this before making the negative assertions that follow
+    cy.get("editor-wc").shadow().contains("Text output").should("be.visible");
+    cy.get("editor-wc").shadow().find("button").contains("Run").should("be.visible");
+
+    // Check that none of the other editor components are displayed in output-only mode
+    cy.get("editor-wc").shadow().contains("Project files").should("not.exist");
+    cy.get("editor-wc").shadow().contains("Don't Collide: Clean Car").should("not.exist");
+    cy.get("editor-wc").shadow().contains("The draw_obstacle function goes here").should("not.exist");
+
+    // Run the code and check it executed without error
+    cy.get("editor-wc").shadow().find("button").contains("Run").click();
+    cy.get("#results").should("contain", '{"errorDetails":{}}');
+
+    // Check that the visual output panel is displayed in split view mode (vs tabbed view)
+    cy.get("editor-wc").shadow().contains("Visual output").should("be.visible");
+    cy.get("editor-wc").shadow().contains("Visual output").parents("ul").children().should("have.length", 1);
+  });
+
+  it("displays the embedded view for an HTML project", () => {
+    cy.visit(urlFor("anime-expressions-solution"));
+
+    // Check HTML preview output panel is visible and has a run button
+    // Important to wait for this before making the negative assertions that follow
+    cy.get("editor-wc").shadow().contains("index.html preview").should("be.visible");
+    cy.get("editor-wc").shadow().find("button").contains("Run").should("be.visible");
+
+    // Check that the code has automatically run i.e. the HTML has been rendered
+    cy.get("editor-wc").shadow().find("iframe#output-frame").its("0.contentDocument.body").should("contain", "Draw anime with me");
+
+    // Check that none of the other editor components are displayed in output-only mode
+    cy.get("editor-wc").shadow().contains("Project files").should("not.exist");
+    cy.get("editor-wc").shadow().contains("Anime expressions solution").should("not.exist");
+    cy.get("editor-wc").shadow().contains("<h1>Draw anime with me</h1>").should("not.exist");
```

Comment:
I'm not so familiar with cypress but this stands out to me - only because it could be interpreted that it's searching for the string of `<h1>Draw anime with me</h1>` - do we want to be finding a `h1` element that contains the text `Draw anime with me`? Or does this effectively parse the string as a `h1` tag?

Reply:
Bonus thought - I'm mostly just concerned that this passes for the right reasons - if it's searching for the parsed string of `<h1>Draw anime...` it could possibly pass erroneously as we are checking for an absence which will pass much more readily.

Reply:
Ah, yes - I can see how that might be confusing. In this case I want to check the editor input component is present and displaying the code, so the text "<h1>Draw anime with me</h1>" is actually visible to the user including the "<h1>" bits! Maybe I should add a comment to explain that a bit...?

Reply:
@conorriches I've tried to address this in https://github.com/RaspberryPiFoundation/editor-ui/pull/1030/commits/58276a3cce45be82cdd5ffdc59bb48ebafd735e by adding comments like "Check that the editor input containing the code is not displayed". Does that help?

Reply:
> Bonus thought - I'm mostly just concerned that this passes for the right reasons - if it's searching for the parsed string of `<h1>Draw anime...` it could possibly pass erroneously as we are checking for an absence which will pass much more readily.

Yeah - I did check that it fails if I do not set `output_only` to `true`. Also I'd like to think there are other e2e specs that would fail if the source code was not displayed when `output_only` is not set.

Reply:
Fab - thanks @floehopper ! That clarifies it nicely